front-end/src/components/Post/GovernanceSideBar/index.tsx

```diff
@@ -55,6 +56,18 @@ const GovenanceSideBar = ({ className, isProposal, isReferendum, onchainId, stat
 	const onAccountChange = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
 		const addressValue = data.value as string;
 		setAddress(addressValue);
+
+		if (!api) {
+			console.error('polkadot/api not set');
+			return;
+		}
+
+		api.query.balances
+			.freeBalance(addressValue, (currentBalance) => {
+				setBalance(currentBalance.toString());
+			})
+			.then(unsubscribe => { unsubscribe(); })
+			.catch(console.error);
```

Comment:
Subscribing to unsubscribe in the `then` is prob. not the way it should be used, also there another such code in the same component.
`Balance` should be it's own component, with address and api as a prop and we should subscribe in a `useEffect` to the balance, and return unsubscribe (to unsubscribe when the component unmounts). This way this component will be isolated and re-render whenever a new address is selected.

BTW we should at some point put `api` as a custom hook ([as done here](https://github.com/substrate-developer-hub/substrate-front-end-template/tree/master/src/substrate-lib)) to avoid props drilling).

Reply:
ah very nice observation. Moved blance to seperate component. Much cleaner